src/borg/remote.py

```diff
@@ -306,6 +316,26 @@ def ssh_cmd(self, location):
             args.append('%s' % location.host)
         return args
 
+    def write_bandwith_limited(self, to_send):
+        if self.ratelimit:
+            current = time.monotonic()
+            if self.ratelimit_last + RATELIMIT_PERIOD <= current:
+                self.ratelimit_quota += self.ratelimit
+                if self.ratelimit_quota > 2 * self.ratelimit:
+                    self.ratelimit_quota = 2 * self.ratelimit
+                self.ratelimit_last = current
+            if self.ratelimit_quota == 0:
+                tosleep = self.ratelimit_last + RATELIMIT_PERIOD - current
+                time.sleep(tosleep)
+                self.ratelimit_quota += self.ratelimit
+                self.ratelimit_last = time.monotonic()
+            if len(to_send) > self.ratelimit_quota:
+                to_send = to_send[:self.ratelimit_quota]
+        written = os.write(self.stdin_fd, to_send)
+        if written > 0 and self.ratelimit:
```

Comment:
Yes, i guess python handles are errors via exceptions, i was thinking to much like raw C.